### PR TITLE
[CHK-1565] feat(tracing): add tracing to retry services

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* feat(tracing): add `TracedRetryEventService` to `RefundRetryService`
* feat(tracing): add `TracedRetryEventService` to `NotificationRetryService`
* feat(tracing): use `TracedRetryEventService` for `ClosureRetryService`
* rename `TracedRetryEventService` to `RetryEventService`

#### Motivation and Context

This PR adds tracing context propagation to retry services, and allows correlating retries with the original processes that triggered them.
Combined with the previous PRs regarding tracing, this PR allows end-to-end correlation from the original producer to both the first consumer and later retries.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

Depends on #108, #130
Depends on #113
Depends on #115
Depends on #116
Depends on #117
Depends on #119
Depends on #126
Depends on #131

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -479,8 +519,8 @@
 				<configuration>
 					<skipCheckoutIfExists>false</skipCheckoutIfExists>
 					<connectionUrl>scm:git:https://github.com/pagopa/pagopa-ecommerce-commons.git</connectionUrl>
-					<scmVersionType>tag</scmVersionType>
-					<scmVersion>${pagopa-ecommerce-commons.version}</scmVersion>
+					<scmVersionType>branch</scmVersionType>
```
Comment (pietro-tota) [inline pom.xml:522]: to be restored

**Review by pietro-tota — DISMISSED** [review]

**Review by pietro-tota — DISMISSED** [review]

File: src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionExpirationQueueConsumerTests.kt [hunk 2]
```diff
@@ -145,6 +146,40 @@ class TransactionExpirationQueueConsumerTests {
     verify(checkpointer, Mockito.times(1)).success()
   }
 
+  @Test
+  fun `messageReceiver receives legacy activated messages successfully`() {
```
Comment (pietro-tota) [inline src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionExpirationQueueConsumerTests.kt:150]: Could you add a test against legacy transaction expired event too?

**Review by pietro-tota — APPROVED** [review]
Well done, just a comment

File: src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionExpirationQueueConsumerTests.kt (hunk 2 above)
Comment (giovanniberti) [reply src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/TransactionExpirationQueueConsumerTests.kt:150]: Fixed in 342d9d07c4afe10f5223847612fca66c26476bc6

**Pull request merged**